src/glove.c

```diff
@@ -63,23 +63,24 @@ int scmp( char *s1, char *s2 ) {
 }
 
 void initialize_parameters() {
-	long long a, b;
-	vector_size++; // Temporarily increment to allocate space for bias
-    
-	/* Allocate space for word vectors and context word vectors, and correspodning gradsq */
-	a = posix_memalign((void **)&W, 128, 2 * vocab_size * (vector_size + 1) * sizeof(real)); // Might perform better than malloc
-    if (W == NULL) {
-        fprintf(stderr, "Error allocating memory for W\n");
-        exit(1);
-    }
-    a = posix_memalign((void **)&gradsq, 128, 2 * vocab_size * (vector_size + 1) * sizeof(real)); // Might perform better than malloc
-	if (gradsq == NULL) {
-        fprintf(stderr, "Error allocating memory for gradsq\n");
-        exit(1);
-    }
-	for (b = 0; b < vector_size; b++) for (a = 0; a < 2 * vocab_size; a++) W[a * vector_size + b] = (rand() / (real)RAND_MAX - 0.5) / vector_size;
-	for (b = 0; b < vector_size; b++) for (a = 0; a < 2 * vocab_size; a++) gradsq[a * vector_size + b] = 1.0; // So initial value of eta is equal to initial learning rate
-	vector_size--;
+  long long a, b;
+
+  /* Allocate space for word vectors and context word vectors, and correspodning gradsq */
+  long long w_size = (2 * vocab_size * (vector_size+2) + vector_size );
```

Comment:
I agree that in general one loop is cleaner than two loops for this code.  The "temporarily increment to allocate..." is a little weird too.  However, isn't the right size 2 * vocab_size * (vector_size + 1)?  Ithink there are some extra factors of vocab_size and vector_size in this expression.  Feel free to correct me as I am not 100% familiar with this code yet.

Reply:
well, it is 3 years ago now, so I am as unfamiliar with the code as you.
We did have a use case that triggered overflows, but I don't have it (and the Colleague)  around anymore. So let's leave it for now